.travis.yml

```diff
@@ -6,19 +6,19 @@ rvm:
   - 2.4.2
   - 2.5.0
   - ruby-head
-  - jruby-head
 
 sudo: false
 
 script:
   - bundle exec rake code_analysis
   - bundle exec rspec
 
+before_install:
+  - gem install sqlite3 -v '1.3.13' --source 'https://rubygems.org/'
+
 matrix:
-  rvm:
-    - 2.4.2
-    - 2.5.0
-    - ruby-head
-    - jruby-head
-  gemfile:
-    - gemfiles/RailsHeadGemfile
+  include:
```

Comment:
maybe it's better to leave it as it was and work with `exclude` instead of `include`. What are you trying to solve here?

Reply:
So, the idea was having an automated test that used Rails head to check for any possible integration errors with newest versions of Rails.

Rails 5.2.0 requires ruby version >= 2.2.2, therefore I am checking with some subsequent versions of ruby, therefore the following part of the document:
```yaml
rvm:
  - 2.2.8
  - 2.3.5
  - 2.4.2
  - 2.5.0
  - ruby-head
```
this ensures that the build is going to run for each of these ruby versions. Now, after testing this with Rails 5.2.0, we needed a

Reply:
So, the idea was having an automated test that used Rails head to check for any possible integration errors with newest versions of Rails.

Rails 5.2.0 requires ruby version >= 2.2.2, therefore I am checking with some subsequent versions of ruby, therefore the following part of the document:
```yaml
rvm:
  - 2.2.8
  - 2.3.5
  - 2.4.2
  - 2.5.0
  - ruby-head
```
which creates a 5x1 matrix (since there are 5 ruby versions and only one gemfile). This ensures that the build is going to run for each of these ruby versions.
Now, after testing this with Rails 5.2.0, we needed a way to test with Rails head. Since [Rails head requires Ruby version >= 2.5.0](https://github.com/rails/rails/blob/master/rails.gemspec) I had to include this in a different matrix, therefore I defined the following matrix:
```yaml
matrix:
  include:
  - rvm: 2.5.0
    gemfile: gemfiles/RailsHeadGemfile
  - rvm: ruby-head
    gemfile: gemfiles/RailsHeadGemfile
```
so Travis includes this gemfile into it's tests. Without this matrix specified it wouldn't test against Rails head.
I don't think using exclude would work, since what this does is exclude a particular build from the matrix as specified on their (docs)[https://docs.travis-ci.com/user/build-matrix#excluding-jobs]. Since the original matrix defined (5x1) does not need any exclusion (all five ruby versions have to be tested) I don't believe using exclude would work here.
The usage of include is so it adds the 2x1 matrix (ruby 2.5.0 and ruby-head with the gemfile that tests rails head) to the already existing 5x1 matrix.

Reply:
Should we remove this 2x1 matrix from here and add it on the future?